### PR TITLE
feat(object-search): individual space support global object search

### LINKED ISSUE #2432: [Feature]: Personal space support global object search
### Is your feature request related to a problem?

Personal space does not support global object search.

### Describe the solution you'd like

Personal space support global object search.

### Additional context

_No response_

### LINKED ISSUE #2443: [Bug]: Database object retrieval results not sorted by match level
### ODC version

4.3.0

### OB version

Not matter.

### What happened?

image
The result list is not sorted by matching level.

### What did you expect to happen?

The result list is sorted by matching level.

### How can we reproduce it (as minimally and precisely as possible)?

.

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-feature
module-global object search

#### What this PR does / why we need it:
This PR add support for global object search:
1. Add system config `odc.database.schema.global-search.enabled` to control the database object sync, the default value  is `true`.
2. Add a user config `odc.database.default.enableGlobalObjectSearch` to control whether open the global object search UI, the default value is `true`.
3. For web mode, if the system config value is `false`, then all sync operation will be ignored; if it is `true`, then we will only sync  databases in team space and these in individual space that the user config is `true`.
4. For desktop mode, if user config is `false`, then sync will be ignored.

#### Which issue(s) this PR fixes:

Fixes #2432 
Fixes #2443 

#### Special notes for your reviewer:
Self-test passed.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java [hunk 1]
```diff
@@ -153,7 +157,7 @@ public QueryDBObjectResp listDatabaseObjects(@NonNull @Valid QueryDBObjectParams
             resp.setDatabases(databases.stream().filter(e -> e.getName().toLowerCase().contains(key))
                     .collect(Collectors.toList()));
         }
-        Pageable pageable = PageRequest.of(0, 1000, Sort.by(Direction.DESC, DBColumnEntity_.DATABASE_ID));
+        Pageable pageable = PageRequest.of(0, MAX_SEARCH_SIZE, Sort.by(Direction.DESC, DBColumnEntity_.DATABASE_ID));
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java:160]: 1000 --> 10000? Is this change expected?

File: server/odc-service/src/main/resources/config-meta/user-config-meta.yml [hunk 2]
```diff
@@ -170,4 +170,14 @@
   allowed_values: [ 'true','false' ]
   default_value: 'true'
   nullable: false
-  description: 'Whether the query is being rewritten in ob-oracle mode and increases the RowId.'
\ No newline at end of file
+  description: 'Whether the query is being rewritten in ob-oracle mode and increases the RowId.'
+- id: 18
+  category: 'user'
+  key: 'odc.database.default.enableGlobalObjectSearch'
```
Comment (MarkPotato777) [inline server/odc-service/src/main/resources/config-meta/user-config-meta.yml:176]: The key is not the same as the system config key, is it expected?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java (hunk 1 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java:160]: I modified it from 10000 to 5000, which means that we will query at most 5000 records. It is excepted and has been tested. The time cost is about 200ms.

File: server/odc-service/src/main/resources/config-meta/user-config-meta.yml (hunk 2 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/resources/config-meta/user-config-meta.yml:176]: This is user config, it is different from system config. It is excepted.

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]

**Review by PeachThinking — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**